### PR TITLE
templates: add display of citations

### PR DESCRIPTION
**Pull request by tomaszgy**
* Introduces display of citations in detailed view of the record.

* Adds ordering by citation_count in search view.

* Adds small enhancements to display of references in detailed view of the record.

Signed-off-by: Tomasz <tomaszgy@gmail.com>

### CODE REVIEW

**Comment by kaplun** [comment]
Uh oh :) some crash with some records...

**Comment by tomaszgy** [comment]
I think we are missing this: https://github.com/inspirehep/invenio-records/pull/8.

**Comment by jmartinm** [comment]
Now that I merged  inspirehep/invenio-records#8, can you rebase this branch with latest `master` so that Travis runs again.

**Comment by tomaszgy** [comment]
Done. Hope it'll go green :smile:

File: inspirehep/utils/citations.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        citations = es_query.records()
+
+        for index, citation in enumerate(citations):
+            out += render_template_to_string("references.html",
+                                             number=str(index + 1),
+                                             record=citation)
+        if out:
+            return out
+        else:
+            return 'There are no citations available for this record'
```
Comment (kaplun) [inline inspirehep/utils/citations.py]: Shouldn't this be templated somehow?

Comment (tomaszgy) [inline inspirehep/utils/citations.py]: You mean all the lines you quoted?

**Comment by tomaszgy** [comment]
PEP-8 error corrected.

File: inspirehep/ext/jinja_filters/record.py [hunk 2]
```diff
@@ -407,3 +418,14 @@ def ads_links(record):
         else:
             link = "%sauthor=%s" % (ADSURL, record['name']['preferred_name'])
         return link
+
+    @app.template_filter
+    def citation_count(record):
```
Comment (jmartinm) [inline inspirehep/ext/jinja_filters/record.py]: Is this filter being used?

It seems like you are accessing directly the value in your templates without the need for a filter.

File: inspirehep/ext/jinja_filters/record.py [hunk 3]
```diff
@@ -407,3 +418,14 @@ def ads_links(record):
...
+        return record.get('citation_count')
+
+    @app.template_filter()
+    def citation_phrase(count):
```
Comment (jmartinm) [inline inspirehep/ext/jinja_filters/record.py]: I don't see this one being used in this PR either?

File: inspirehep/utils/citations.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+from invenio.ext.template import render_template_to_string
```
Comment (jmartinm) [inline inspirehep/utils/citations.py]: The correct import is:

```python
from invenio_ext.template import render_template_to_string
```

**Comment by jmartinm** [comment]
For the search sorting to work you need to change in `inspirehep/base/static/js/search/search_results.js`:

```diff
-           case 'oldest':
-            $sortField.val('earliest_date');
-            $sortOrder.val('asc');
+          case 'mostcited':
+            $sortField.val('citation_count');
+            $sortOrder.val('desc');
```

**Comment by jmartinm** [comment]
And please, rebase this branch against latest master

File: inspirehep/ext/jinja_filters/record.py (hunk 2 above)
Comment (tomaszgy) [inline inspirehep/ext/jinja_filters/record.py]: You're right. Just removed it.

File: inspirehep/ext/jinja_filters/record.py (hunk 3 above)
Comment (tomaszgy) [inline inspirehep/ext/jinja_filters/record.py]: It's now used in the new version of this PR.

File: inspirehep/utils/citations.py (hunk 4 above)
Comment (tomaszgy) [inline inspirehep/utils/citations.py]: Corrected.

**Comment by jmartinm** [comment]
Continues here https://github.com/inspirehep/inspire-next/pull/702

**Comment by jmartinm** [comment]
Merged https://github.com/inspirehep/inspire-next/pull/702

**Pull request closed without merging**